src/main/java/sg/ncl/MainController.java

```diff
@@ -3312,49 +3313,107 @@ public String adminUsersManagement(Model model, HttpSession session) {
-            end = endDate.format(formatter);
+            end = now;
+        }
+        if (now.compareTo(start) < 0 || now.compareTo(end) < 0) {
+            redirectAttributes.addFlashAttribute(MESSAGE, "Period selected is beyond current date (today).");
+            return REDIRECT_TEAM_USAGE;
         }
 
         // get list of teamids
         HttpEntity<String> request = createHttpEntityHeaderOnly();
         ResponseEntity responseEntity = restTemplate.exchange(properties.getSioTeamsUrl(), HttpMethod.GET, request, String.class);
         JSONArray jsonArray = new JSONArray(responseEntity.getBody().toString());
 
+        List<Team2> searchTeams = new ArrayList<>();
         TeamManager2 teamManager2 = new TeamManager2();
-        for (int i = 0; i < jsonArray.length(); i++) {
-            JSONObject jsonObject = jsonArray.getJSONObject(i);
-            Team2 one = extractTeamInfo(jsonObject.toString());
-            teamManager2.addTeamToTeamMap(one);
-        }
+        getSearchTeams(organizationType, team, jsonArray, searchTeams, teamManager2);
 
-        if (team != null) {
-            responseEntity = restTemplate.exchange(properties.getUsageStat(team, "startDate=" + start, "endDate=" + end), HttpMethod.GET, request, String.class);
-            String usage = responseEntity.getBody().toString();
-            model.addAttribute("usage", usage);
+        if (!searchTeams.isEmpty()) {
+            List<String> dates = getDates(start, end, formatter);
+
+            Map<String, List<Long>> teamUsages = new HashMap<>();
+            Long totalUsage = 0L;
+            for (Team2 team2 : searchTeams) {
+                try {
+                    List<Long> usages = new ArrayList<>();
+                    totalUsage += getTeamUsageStatistics(team2, start, end, request, usages);
+                    teamUsages.put(team2.getName(), usages);
+                } catch (RestClientException rce) {
+                    log.warn("Error connecting to sio analytics service for team usage: {}", rce);
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+                    return REDIRECT_TEAM_USAGE;
+                } catch (StartDateAfterEndDateException sde) {
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERR_START_DATE_AFTER_END_DATE);
+                    return REDIRECT_TEAM_USAGE;
+                }
+            }
+            model.addAttribute("dates", dates);
+            model.addAttribute("teamUsages", teamUsages);
+            model.addAttribute("totalUsage", totalUsage);
         }
 
         model.addAttribute("teamsMap", teamManager2.getTeamMap());
         model.addAttribute("start", start);
         model.addAttribute("end", end);
+        model.addAttribute("organizationType", organizationType);
         model.addAttribute("team", team);
         return "usage_statistics";
     }
 
+/*
+    private String getStartDate(DateTimeFormatter formatter, ZonedDateTime nowDate) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)